### PR TITLE
Task 51.1: BE: Cart: Stripe integration to complete order

### PR DESCRIPTION
**Pull request by di-kinosian**

### CODE REVIEW

File: src/cart/cart.service.ts [hunk 1]
```diff
@@ -21,10 +21,15 @@ export class CartService {
...
+    let cart = await this.cartModel.findOne({ userId });
 
     if (!cart) {
-      throw new NotFoundException(ErrorMessages.CART_NOT_FOUND);
+      cart = new this.cartModel({
+        userId,
+        items: [],
+      });
+      await cart.save();
+      return [];
```
Comment (VYaniv) [inline src/cart/cart.service.ts:33]: Empty line before return

File: src/order/order.service.ts [hunk 2]
```diff
@@ -33,6 +41,44 @@ export class OrderService {
...
+
+    await createdOrder.save();
+
+    return {
+      url: session.url,
+      order: createdOrder,
+    };
+  }
+
+  async confirmOrder(userId: string, sessionId: string) {
```
Comment (VYaniv) [inline src/order/order.service.ts]: Returned type

File: src/order/order.service.ts [hunk 3]
```diff
@@ -33,6 +41,44 @@ export class OrderService {
...
+    let order = await this.orderModel.findOne({
+      userId,
+      checkoutId: sessionId,
+    });
+
+    if (order) {
+      await this.cartModel.updateOne({ userId }, { $unset: { items: [] } });
+      order.status = OrderStatus.COMPLETED;
+      return await order.save();
```
Comment (VYaniv) [inline src/order/order.service.ts:90]: Empty line before return

Comment (VYaniv) [reply src/order/order.service.ts:90]: Don`t we need the response if there is no order?

**Review by VYaniv — APPROVED** [review]

File: src/order/order.service.ts (hunk 3 above)
Comment (di-kinosian) [reply src/order/order.service.ts:90]: @VYaniv , yep, I agree, we need to return an error if the order is not found

File: src/cart/cart.service.ts (hunk 1 above)
Comment (di-kinosian) [reply src/cart/cart.service.ts:33]: done

File: src/order/order.service.ts (hunk 2 above)
Comment (di-kinosian) [reply src/order/order.service.ts]: done

**Pull request merged**